src/dns/error.cpp

```diff
@@ -0,0 +1,86 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/common/error.hpp>
+#include <measurement_kit/dns/error.hpp>
+
+#include <event2/dns.h>
+
+namespace measurement_kit {
+namespace dns {
+
+common::Error dns_error(int code) {
```

Comment:
It was removed because I migrated the code from `[](dns::Response resp) {...}` to `[](Error error, dns::Response resp) {...}`. This migration was done for consistency with other places in the code, and because I think it is cleaner to separate error and response.

Because of this change it was natural to move the conversion to OONI error in the Error class. Since also other parts of MK will need to map errors to OONI errors, furthermore, it seems to me more uniform that one always calls `error.as_ooni_error()` to access the OONI error bound to a specific MK error.